2022/02/ChAoS-UnItY/src/main.rs

```diff
@@ -0,0 +1,66 @@
+use std::collections::HashMap;
+
+macro_rules! hashmap {
+    ($( $key: expr => $val: expr ),*) => {{
+         let mut map = HashMap::new();
+         $( map.insert($key, $val); )*
+         map
+    }}
+}
+
+fn main() {
+    // Pre-computed
+    let map = hashmap!(
+        'A' => hashmap!(
+            'X' => (3 + 1, 3 + 0),
+            'Y' => (6 + 2, 1 + 3),
+            'Z' => (0 + 3, 2 + 6)
+        ),
+        'B' => hashmap!(
+            'X' => (0 + 1, 1 + 0),
+            'Y' => (3 + 2, 2 + 3),
+            'Z' => (6 + 3, 3 + 6)
+        ),
+        'C' => hashmap!(
+            'X' => (6 + 1, 2 + 0),
+            'Y' => (0 + 2, 3 + 3),
+            'Z' => (3 + 3, 1 + 6)
+        )
+    );
+
+    let data = include_str!("../Day02.txt");
+    let processed_data = process_data(data);
+
+    println!("{}", part1(&processed_data, &map));
+    println!("{}", part2(&processed_data, &map));
+}
+
+fn part1(data: &Vec<(char, char)>, map: &HashMap<char, HashMap<char, (i32, i32)>>) -> i32 {
```

Comment:
A common pattern is to prefer slice type over concrete `Vec` type. Slice (`&[T]`) is usually more useful and flexible than an `&Vec<T>`.

```suggestion
fn part1(data: &[(char, char)], map: &HashMap<char, HashMap<char, (i32, i32)>>) -> i32 {
```